lib/vrt/map.rb

```diff
@@ -11,25 +11,22 @@ class Map
     def initialize(version = nil)
       @version = version || VRT.current_version
       @structure = build_structure
-      @found_nodes = {}
-      @lineages = {}
+      @_found_nodes = {}
+      @_lineages = {}
+      @_valid_vrt_ids = {}
+      @_valid_identifiers = {}
     end
 
     def find_node(string, max_depth: 'variant')
-      @found_nodes[string + max_depth] ||= walk_node_tree(string, max_depth: max_depth)
+      @_found_nodes[string + max_depth] ||= walk_node_tree(string, max_depth: max_depth)
     end
 
-    def valid?(node)
-      return true if node == 'other'
-      # At least one string of lowercase or _, plus up to 2 more with stops
-      return false unless node =~ /\A[a-z_]+(\.[a-z_]+){0,2}\z/
-      found_node = find_node(node)
-      return false unless found_node
-      true
+    def valid?(vrt_id)
+      @_valid_vrt_ids[vrt_id] ||= valid_identifier?(vrt_id) && find_node(vrt_id)
```

Comment:
This is returning nil, or a found node. I think by convention we should be coercing the result to a boolean

Reply:
doesn't the && coerce it into a bool?

Reply:
no, you're right, when I was looking at this, rubocop was very offended by `!!` but it can be acheived in other ways

Reply:
my b, I'm still learning ruby lol.

Reply:
lol you don't know how many times I run that locally in both javascript and ruby and question my existence